appsec/src/extension/commands/request_shutdown.c

```diff
@@ -189,3 +274,393 @@ static void _pack_headers_no_cookies_map(
     ZEND_HASH_FOREACH_END();
     mpack_finish_map(w);
 }
+
+static const char *nullable _header_content_type_zend_array(
+    const zend_array *nonnull hl, size_t *nonnull len)
+{
+
+    zend_string *key;
+    zval *val;
+    zend_long idx;
+    ZEND_HASH_FOREACH_KEY_VAL((zend_array *)hl, idx, key, val)
+    {
+        (void)idx;
+        if (!key) {
+            continue;
+        }
+        if (zend_string_equals_literal_ci(key, "content-type")) {
+            if (Z_TYPE_P(val) == IS_STRING) {
+                *len = Z_STRLEN_P(val);
+                return Z_STRVAL_P(val);
+            }
+            if (Z_TYPE_P(val) == IS_ARRAY) {
+                zend_array *arr = Z_ARR_P(val);
+                HashPosition pos;
+                zend_hash_internal_pointer_end_ex(arr, &pos);
+                zval *zv = zend_hash_get_current_data_ex(arr, &pos);
+                if (!zv) {
+                    continue;
+                }
+                ZVAL_DEREF(zv);
+                if (Z_TYPE_P(zv) != IS_STRING) {
+                    continue;
+                }
+                *len = Z_STRLEN_P(zv);
+                return Z_STRVAL_P(zv);
+            }
+        }
+    }
+    ZEND_HASH_FOREACH_END();
+
+    return NULL;
+}
+
+static zval _convert_json(char *nonnull entity, size_t entity_len)
+{
+    zval zv;
+    ZVAL_NULL(&zv);
+    if (!_json_decode_ex) {
+        return zv;
+    }
+
+#define MAX_DEPTH 30
```

Comment:
Schema extraction has a depth limit of 18, while the scalar rules have one of 20, so 30 seems overkill.

Reply:
I don't think it's overkill, because it will allow for more data to be partially sent to the waf. If the limit here is exceeded, no data will be sent.

Reply:
Right, that's fair.